src/c/common.c

```diff
@@ -177,7 +181,7 @@ size_t ucdr_buffer_alignment(
         size_t data_size)
 {
     return (data_size > ub->last_data_size)
-        ? (data_size - ((uint32_t)(ub->offset - ub->origin) % data_size)) & (data_size - 1)
+        ? (data_size - ((uint32_t)(ub->offset) % data_size)) & (data_size - 1)
```

Comment:
Is `ub->offset` enough to detect the current alignment?

Reply:
We should check if `offset` is the amount of serialized bytes in total or the serialized in the current buffer.

Reply:
According to the [Readme](https://github.com/eProsima/Micro-CDR/blob/master/README.md):
> - `origin`: the origin of the XCDR stream.
> - `offset`: where the serialization/deserialization will start.
> Initially, the serialization/deserialization starts at the beginning of the buffer.